docs/cpp.rst

```diff
@@ -0,0 +1,52 @@
+.. _cccl-cpp-libraries:
+
+CCCL C++ Libraries
+==================
```

Comment:
suggestion: CCCL already has Libraries in it, maybe something along the following lines?
```suggestion
CUDA C++ Core Libraries
=======================
```